src/uu/numfmt/src/numfmt.rs

```diff
@@ -166,6 +180,38 @@ struct NumfmtOptions {
     transform: TransformOptions,
     padding: isize,
     header: usize,
+    fields: Vec<Range>,
+}
+
+/// Iterate over a line's fields, where each field is a contiguous sequence of
+/// non-whitespace, optionally prefixed with one or more characters of leading
+/// whitespace. Fields are returned as tuples of `(prefix, field)`.
+struct WhitespaceSplitter<'a> {
+    s: Option<&'a str>,
+}
+
+impl<'a> Iterator for WhitespaceSplitter<'a> {
+    type Item = (&'a str, &'a str);
+
+    fn next(&mut self) -> Option<Self::Item> {
```

Comment:
could you please add a comment for this function? thanks

Reply:
done! :)